src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -283,13 +259,13 @@ private IList<ContentItem> FindItemsImplementation(PatternSet definition, IEnume
                                 contentItem.Properties.Add("related", relatedFileExtensionsProperty);
```

Comment:
I'm uncertain about how this PR addresses the allocations problem mentioned in the work item https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1837276. The bug's call stack indicated a resize operation in the Dictionary, and as far as I comprehend, we haven't made any modifications to that specific part. I believe `contentItem.Properties.Add("related", relatedFileExtensionsProperty);` line of code is causing the dictionary resizing.

Reply:
Was that the right bug? This appears to have implemented exactly what is called in the bug you listed.

Reply:
I think the confusion here is because the discussion in that bug talks about `List<>` but the callstack in that bug talks about `Dictionary<>`.

The list allocations have been addressed here. The dictionary one has not, as far as I can tell.

I'll pull down the `dev` branch now that this has merged to see whether there's more we can do here.

@davkean I'll leave it to you to decide whether this closes that bug or not.

Reply:
Okay, I understand. The call stack is an example call stack from that failure - there are many in this case. In this case I looked at the trace and determined that there was easy fixable path and opened a bug from the failure, the tooling randomly picked a stack and put that in the bug which didn't match what I wanted to get fixed.

Reply:
> I'll pull down the dev branch now that this has merged to see whether there's more we can do here.

I can't see an obvious way forward here. It doesn't seem possible to predict the collection size in advance.

Reply:
Yeah I think I made a similar conclusion as I didn't file a bug here despite the resize being obvious in the stack.